### PR TITLE
Complexity. Displayed below the top block (similar to block comments)

### PR DESCRIPTION
**Pull request by sdh11**
Added the complexity measure for a flowgraph. Value is now shown below the top block in a similar fashion to block comments. Can be enabled and disabled from the Tools menu.

### CODE REVIEW

File: grc/gui/ActionHandler.py [hunk 1]
```diff
@@ -127,6 +127,7 @@ def _handle_action(self, action):
                 Actions.TOGGLE_AUTO_HIDE_PORT_LABELS, Actions.TOGGLE_SNAP_TO_GRID,
                 Actions.TOGGLE_SHOW_BLOCK_COMMENTS,
                 Actions.TOGGLE_SHOW_CODE_PREVIEW_TAB,
+                Actions.TOGGLE_SHOW_FLOWGRAPH_COMPLEXITY,
```
Comment (skoslowski) [inline grc/gui/ActionHandler.py]: You'll have to add this to the list of actions to load from prefs. at line 146. Better yet we can get rid of that for loop there and add ``` if hasattr(action, 'load_from_preferences'): action.load_from_preferences() ``` to the loop above.

File: grc/gui/Block.py [hunk 2]
```diff
@@ -39,6 +40,15 @@
 #set $foreground = $block.get_enabled() and '#444' or '#888'
 <span foreground="$foreground" font_desc="$font">$encode($block.get_comment())</span>"""
 
+# Includes the additional complexity markup if enabled
+COMPLEXITY_MARKUP_TMPL="""\
+<span foreground="#444" size="medium" font_desc="$font"><b>$encode("Complexity: {}bal".format($complexity))</b></span>"""
+COMMENT_COMPLEXITY_MARKUP_TMPL="""\
+#set $foreground = $block.get_enabled() and '#444' or '#888'
+<span foreground="#444" size="medium" font_desc="$font"><b>$encode("Complexity: {}bal".format($complexity))</b></span>
```
Comment (skoslowski) [inline grc/gui/Block.py]: A template within a template =)
Use one template. Pass ```complexity=None``` to hide it and you can get rid of that logic in create_comment_label() altogether.

File: grc/gui/Block.py [hunk 3]
```diff
@@ -273,9 +304,10 @@ def what_is_selected(self, coor, coor_m=None):
...
-
+        # Show the flowgraph complexity in the same location as comments.
         x, y = self.get_coordinate()
-        window.draw_drawable(gc, self._comment_pixmap, 0, 0, x,
-                             y + self.H + BLOCK_LABEL_PADDING, -1, -1)
+        comment_height = 0
+        if self._comment_pixmap:
+            window.draw_drawable(gc, self._comment_pixmap, 0, 0, x,
+                                 y + self.H + BLOCK_LABEL_PADDING, -1, -1)
+            comment_height = self._comment_pixmap.get_size()[1] + 5
```
Comment (skoslowski) [inline grc/gui/Block.py]: What do you need the comment_height for?

File: grc/gui/Constants.py [hunk 4]
```diff
@@ -59,7 +59,7 @@
 
 #  flow-graph canvas fonts
 try:  # ugly, but matches current code style
-    FONT_SIZE = _gr_prefs.get_long('grc', 'canvas_font_size', 8)
+    FONT_SIZE = _gr_prefs.get_long('grc', 'canvas_font_size', 10)
```
Comment (skoslowski) [inline grc/gui/Constants.py]: Why change that. (The prefs have 8 set by default anyway)

File: grc/python/Param.py [hunk 5]
```diff
@@ -88,18 +94,7 @@ def _truncate(string, style=0):
...
-        ##################################################
-        def num_to_str(num):
-            if isinstance(num, COMPLEX_TYPES):
-                num = complex(num) #cast to python complex
-                if num == 0: return '0' #value is zero
-                elif num.imag == 0: return '%s'%eng_notation.num_to_str(num.real) #value is real
-                elif num.real == 0: return '%sj'%eng_notation.num_to_str(num.imag) #value is imaginary
-                elif num.imag < 0: return '%s-%sj'%(eng_notation.num_to_str(num.real), eng_notation.num_to_str(abs(num.imag)))
-                else: return '%s+%sj'%(eng_notation.num_to_str(num.real), eng_notation.num_to_str(num.imag))
-            else: return str(num)
```
Comment (skoslowski) [inline grc/python/Param.py]: I kind excepted some usages of this function to change? Aren't there any?

File: grc/gui/Utils.py [hunk 6]
```diff
@@ -21,6 +21,8 @@
 pygtk.require('2.0')
 import gtk
 import gobject
+import numpy
+from gnuradio import eng_notation
```
Comment (skoslowski) [inline grc/gui/Utils.py]: ?

**Comment by skoslowski** [comment]
Please always double-check the complete diff of your PR (against its base) after making changes. This way you easily spot obsolete changes that remain after editing the original PR with additional commits.

File: grc/python/Param.py (hunk 5 above)
Comment (sdh11) [inline grc/python/Param.py]: I don't believe so. Since it was nested, the __repr__() function was the only one to ever use it. Since I just moved it to a global declaration in that file, the usage syntax didn't need to be changed.

File: grc/gui/Constants.py (hunk 4 above)
Comment (sdh11) [inline grc/gui/Constants.py]: On OS X, a font size of 8 is really small and difficult to read. I need to make this part of a separate PR anyway.

File: grc/gui/Block.py (hunk 3 above)
Comment (sdh11) [inline grc/gui/Block.py]: Whoops. Forgot to remove that.

**Comment by sdh11** [comment]
Ok. I addressed everything you mentioned, cleaned up the old changes, rebased to master_grcwg, and squashed the commits. I checked over this diff and I think it looks good.

**Comment by skoslowski** [comment]
Great. Thanks. I think this is about ready to be merged. I had three minor changes which I amended your commit. I also added some padding around the comment boxes. See

https://github.com/skoslowski/gnuradio-wg-grc/commits/complexity

I will merge later today if there are no objections

**Comment by sdh11** [comment]
Looks good to me.
